backend/app/notify/email.go

```diff
@@ -227,47 +239,105 @@ func NewEmail(emailParams EmailParams, smtpParams SmtpParams) (*Email, error) {
 	return &res, err
 }
 
-// Send email about reply to Request.Email if it's set, otherwise do nothing and return nil, thread safe
-// do not returns sending error, only following:
-// 1. (likely impossible) template execution error from email message creation from Request
-// 2. message dropped without sending in case of closed ctx
+// Send email about comment reply to Request.Email if it's set,
+// also sends email to site administrator if appropriate option is set.
+// Thread safe
 func (e *Email) Send(ctx context.Context, req Request) (err error) {
```

Comment:
I've made everything non-blocking which solves this problem. I think two emails should be sent from email.Send, rather than notify.Submit knowing anything more about it's destinations.

Reply:
well, I meant something different. Whatever calls `notify.Submit` should do it twice and nothing inside the notify package should be that smart. Generally, the idea of Send sometimes sending one message (email) and sometimes two, feel odd and unexpected

Reply:
We already have to look up user email before sending a notification, which is exposing a few inner details of `Email` to `notify.Submit`.

For admin notifications outside of `Email` module either `notify.Submit` or it's caller need to fetch the admin email and pass it on.
If we want different title \ content of the message for users and for admin, things complicate: we also would need to pass `ForAdmin` flag alongside the message, which would affect its formatting.

It would make some sense to have this complexity exposed to `notify.Submit` or its callers if we would have more notification destinations with similar requirements of sending sometimes user notifications of different kinds and sometimes admin notifications of different kinds.
I would argue that it's better to contain this logic inside `Email` for now as it's the only module that is aware of admin notifications existence and change it when we'll have at least one more notification destination with feature parity with `Email`.

Reply:
tl;dr: moving sending logic up the call stack is a good feature for supporting N notification providers, but at the moment we have only 1 notification provider utilizing this logic and it would introduce complexity with no value paying for it.

Reply:
I have no clue what complexity you meant. All we need is a check and call near this place https://github.com/umputun/remark42/blob/master/backend/app/rest/api/rest_private.go#L134. This place also seems to be the right one to know about admin's email. To satisfy different formatter we have, we could add `Request.SiteAdmin` or smth like this.